### PR TITLE
Updated example code

### PR DESCRIPTION
**Pull request by nils-a**
Bumped projects to SDK style and target net6.0

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -62,4 +62,6 @@ packages
 Thumbs.db
 
 # NUnit
-TestResult.xml
\ No newline at end of file
+TestResult.xml
+.vs/
+.idea/
```
Comment (augustoproiete) [inline .gitignore]: Move to the `# User-specific files` section above?

File: .gitignore [hunk 2]
```diff
@@ -62,4 +62,6 @@ packages
 Thumbs.db
 
 # NUnit
```
Comment (augustoproiete) [inline .gitignore]: Remove this header now that you're removing `TestResult.xml` and there's no NUnit files being ignored?

File: build.cake [hunk 3]
```diff
@@ -29,34 +26,29 @@ Task("Restore-NuGet-Packages")
...
-    }
-    else
+    DotNetCoreBuild("./src/Example.sln", new DotNetCoreBuildSettings
     {
-      // Use XBuild
-      XBuild("./src/Example.sln", settings =>
-        settings.SetConfiguration(configuration));
-    }
+      Configuration = configuration,
+      NoRestore = true,
```
Comment (augustoproiete) [inline build.cake]: Looks like indentation is off (`2` spaces instead of `4`?)

File: build.cake [hunk 4]
```diff
@@ -29,34 +26,29 @@ Task("Restore-NuGet-Packages")
...
     .IsDependentOn("Build")
     .Does(() =>
 {
-    NUnit3("./src/**/bin/" + configuration + "/*.Tests.dll", new NUnit3Settings {
-        NoResults = true
-        });
+    DotNetCoreTest("./src/Example.sln", new DotNetCoreTestSettings
+    {
+      Configuration = configuration,
+      NoRestore = true,
```
Comment (augustoproiete) [inline build.cake]: Looks like indentation is off (`2` spaces instead of `4`?)

File: src/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<Project>
+    <PropertyGroup>
+        <LangVersion>10</LangVersion>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+</Project>
```
Comment (augustoproiete) [inline src/Directory.Build.props]: Looks like indentation is off (`4` spaces instead of `2` for XML?)

**Review by augustoproiete — APPROVED** [review]
LGTM. Added a couple of suggestions.

File: .gitignore (hunk 1 above)
Comment (nils-a) [reply .gitignore]: 👍

File: .gitignore (hunk 2 above)
Comment (nils-a) [reply .gitignore]: 👍

File: build.cake (hunk 4 above)
Comment (nils-a) [reply build.cake]: I've added a shared `.editorconfig` and reformatted.

File: src/Directory.Build.props (hunk 5 above)
Comment (nils-a) [reply src/Directory.Build.props]: I've added a shared `.editorconfig` and reformatted.

File: build.cake (hunk 3 above)
Comment (nils-a) [reply build.cake]: I've added a shared `.editorconfig` and reformatted.

**Review by augustoproiete — APPROVED** [review]
LGTM!

**Pull request merged**